src/osd/PG.h

```diff
@@ -121,7 +121,7 @@ class PGStateHistory {
 
   void dump(Formatter* f) const;
 
-  string get_current_state() {
+  const std::string get_current_state() {
```

Comment:
Maybe const std::string &?

Reply:
Ugh! Of course...

Reply:
Errr... actually this generates a "returning reference to temporary" warning @athanatos

Reply:
Ah, because of the return "unknown" (it's invoking a std::string(const char *) constructor implicitly).  I think you'll need to make that a static constant of type std::string.

Reply:
OK, I'll look into that. There's something wrong with the json output too so this will need a bit more work?